pkg/testcase/certrotate.go

```diff
@@ -0,0 +1,109 @@
+package testcase
+
+import (
+	"fmt"
+
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestCertRotate() {
+	// Please refer to docs/testing.md for instructions on setup for this test
+
+	cluster := factory.AddCluster(GinkgoT())
+	serverIPs := cluster.ServerIPs
+	agentIPs := cluster.AgentIPs
+	p, _ := shared.GetProduct()
+	var product shared.Product
+	if p == "k3s" {
+		product = shared.K3S
+	} else {
+		product = shared.RKE2
+	}
+
+	certRotate(product, serverIPs)
+
+	product.ManageClusterService(shared.Restart, shared.Agent, agentIPs)
+
+	compareAndVerifyTLSDirContent(product, serverIPs)
+
+}
+
+// certRotate Rotate certificate for etcd only and cp only nodes
+func certRotate(product shared.Product, ips []string) {
+
+	stopError := product.ManageClusterService(shared.Stop, shared.Server, ips)
+	if stopError != nil {
+		shared.ReturnLogError(fmt.Sprintf("Error stopping %s service", product))
+	}
+
+	ip, rotateError := product.CertRotate(ips)
+	if rotateError != nil {
+		shared.ReturnLogError(fmt.Sprintf("Error running certificate rotate for %s service on %s", product, ip))
+	}
+
+	startError := product.ManageClusterService(shared.Start, shared.Server, ips)
+	if startError != nil {
+		shared.ReturnLogError(fmt.Sprintf("Error starting %s service", product))
+	}
+	// Preventative restart - sometimes, start hangs after a cert rotate which gets fixed on a second restart
+	restartError := product.ManageClusterService(shared.Restart, shared.Server, ips)
+	if restartError != nil {
+		shared.ReturnLogError(fmt.Sprintf("Error restarting %s service", product))
+	}
+}
+
+// getExpectedFileNames Used for verification of file list
+func getExpectedFileNames() []string {
+	return []string{"client-ca.crt", "client-ca.key", "client-ca.nochain.crt", "client-supervisor.crt",
+		"client-supervisor.key", "peer-ca.crt", "peer-ca.key", "server-ca.crt", "server-ca.key", "request-header-ca.crt",
+		"request-header-ca.key", "server-ca.crt", "server-ca.key", "server-ca.nochain.crt", "service.current.key", "service.key"}
+}
+
+// verifyIdenticalFiles Verify the actual and expected identical file lists match
+func verifyIdenticalFiles(identicalFileList string) {
```

Comment:
this func and its usage on "func compareAndVerifyTLSDirContent(product shared.Product, ips []string) {
"
you. should abstract it to compare any file, and receive a "filename" as parameter 

and them both should live on aux.go, to be used anywhere else later

Reply:
fixed the name of the functions and moved a genric provide filenames actual vs expected given a path to the aux.go. the comparison of tls dir content is very specific to cert rotate - so retaining that portion in the testcase file.